src/sentry/static/sentry/app/types/index.tsx

```diff
@@ -381,6 +381,11 @@ export type PluginWithProjectList = PluginNoProject & {
   projectList: PluginProjectItem[];
 };
 
+export type AppOrProviderOrPlugin =
+  | SentryApp
+  | IntegrationProvider
+  | PluginWithProjectList;
```

Comment:
@NisanthanNanthakumar I think we should use the `PluginNoProject` instead `PluginWithProjectList` if we can here

Reply:
<img width="1440" alt="Screen Shot 2020-03-25 at 12 00 15 PM" src="https://user-images.githubusercontent.com/10491193/77575192-6c53d200-6e90-11ea-86a2-8aef36fce4bb.png">

We have `projectList` in the plugins for Integration Directory. And we use this type `PluginWithProjectList` everywhere to define plugins in Integration directory.

Reply:
@NisanthanNanthakumar The fact that the actual object has the `projectList` field is irrelevant. We should always use the minimalist type possible. For example, we try to use `LightWeightOrganization` over `Organization` whenever possible, even when the actual organization is the latter.

That being said, I believe we can't use `PluginNoProject` because we actually use the `projectList` property in the `integrationListDirectory`. So what you have is fine.